azure-functions-java-sdktypes/src/main/java/com/microsoft/azure/functions/sdktype/blob/BlobContainerHydrator.java

```diff
@@ -0,0 +1,136 @@
+package com.microsoft.azure.functions.sdktype.blob;
+
+import com.microsoft.azure.functions.sdktype.SdkTypeHydrator;
+
+import java.lang.reflect.Method;
+import java.util.logging.Logger;
+
+/**
+ * Reflection logic for building a BlobContainerClient from BlobContainerMetaData,
+ * supporting both connection string usage and managed identity usage.
+ */
+public class BlobContainerHydrator implements SdkTypeHydrator<BlobContainerMetaData> {
+    private static final Logger LOGGER = Logger.getLogger(BlobContainerHydrator.class.getName());
+
+    @Override
+    public Object createInstance(BlobContainerMetaData metaData) throws Exception {
+        String containerName = metaData.getContainerName();
+        String envVar = metaData.getConnectionEnvVar();
+
+        LOGGER.info("BlobContainerHydrator: Using environment variable name as prefix: " + envVar);
+
+        // Check if environment variable => connection string
+        String maybeConnString = System.getenv(envVar);
+        if (maybeConnString != null && isConnectionString(maybeConnString)) {
+            LOGGER.info("Detected connection string usage from environment variable: " + envVar);
+            return buildWithConnectionString(containerName, maybeConnString);
+        } else {
+            // interpret envVar as prefix for managed identity
+            LOGGER.info("Detected managed identity usage. Prefix: " + envVar);
+
+            String accountName = System.getenv(envVar + "__accountName");
+            String serviceUri = System.getenv(envVar + "__serviceUri");
+            String blobServiceUri = System.getenv(envVar + "__blobServiceUri");
+            String clientId = System.getenv(envVar + "__clientId");
+
+            String endpoint = resolveEndpoint(accountName, serviceUri, blobServiceUri);
+            Object credential = buildManagedIdentityCredential(clientId);
+
+            return buildWithManagedIdentity(containerName, endpoint, credential);
+        }
+    }
+
+    private boolean isConnectionString(String val) {
+        return val.contains("AccountKey=") || val.contains("DefaultEndpointsProtocol=");
+    }
+
+    private Object buildWithConnectionString(String containerName, String connStr) throws Exception {
+        ClassLoader cl = Thread.currentThread().getContextClassLoader();
+        Class<?> builderClass = cl.loadClass("com.azure.storage.blob.BlobContainerClientBuilder");
+        Object builder = builderClass.getDeclaredConstructor().newInstance();
+
+        Method connMethod = builderClass.getMethod("connectionString", String.class);
+        connMethod.invoke(builder, connStr);
+
+        Method contMethod = builderClass.getMethod("containerName", String.class);
+        contMethod.invoke(builder, containerName);
```

Comment:
I do not have an idea yet once we will have some improvement, in next iteration we can introduce ReflectionUtils to optimize these reflections based code and take it out from all the places.